actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -356,6 +359,15 @@ def constraints(options, path_params)
           def dispatcher(raise_on_name_error)
             Routing::RouteSet::Dispatcher.new raise_on_name_error
           end
+
+          def route_source_location
+            if Mapper.route_source_locations
+              ad_dir = File.expand_path("..", __dir__)
+              caller_location = caller_locations.find { |cl| !cl.path.include?(ad_dir) }
+              cleaned_path = Mapper.backtrace_cleaner.clean([caller_location.path]).first
+              "#{cleaned_path}:#{caller_location.lineno}" if cleaned_path
```

Comment:
```suggestion
              action_dispatch_dir = File.expand_path("..", __dir__)
              caller_location = caller_locations.find { |location| !location.path.include?(action_dispatch_dir) }
              cleaned_path = Mapper.backtrace_cleaner.clean([caller_location.path]).first
              "#{cleaned_path}:#{caller_location.lineno}" if cleaned_path
```

Reply:
😅 Thanks! Updated in the last force push